File: Paseo2/src/main/java/modelos/Pedido.java
package modelos;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import utils.EstadoPedido;
import utils.RetiroEntrega;

@Entity
public class Pedido {

	@Id @GeneratedValue(strategy=GenerationType.IDENTITY) 
	private Long id;
	
	private Date fecha;
	
	private Date fechaEntrega;
	
	@ManyToOne
	private Usuario usuario;
	
	@Enumerated(EnumType.ORDINAL) //0,1..
	private RetiroEntrega retiroEntrega;
	
	
    //para que lo cambie el admin, no confirmado es carrito
	@Enumerated(EnumType.ORDINAL) //0,1,2..
	private EstadoPedido estado;
	
	@OneToOne
	private PuntoDeRetiro direccionRetiro;
	
	private String direccionEntrega;
	private BigDecimal montoTotal;
	private String rangoHorario;
	
	@OneToMany(mappedBy = "pedido")
	private List<ProductoEncargado> listaProductos;
	
	@ManyToOne 
	private Ronda ronda;
	
	public Pedido() {
		
	}
	public Pedido(Date fecha, Date fechaEntrega, Usuario usuario, RetiroEntrega retiroEntrega, EstadoPedido estado,
			PuntoDeRetiro direccionRetiro, String direccionEntrega, BigDecimal montoTotal, String rangoHorario,
			List<ProductoEncargado> listaProductos, Ronda ronda) {
		super();
		this.fecha = fecha;
		this.fechaEntrega = fechaEntrega;
		this.usuario = usuario;
		this.retiroEntrega = retiroEntrega;
		this.estado = estado;
		this.direccionRetiro = direccionRetiro;
		this.direccionEntrega = direccionEntrega;
		this.montoTotal = montoTotal;
		this.rangoHorario = rangoHorario;
		this.listaProductos = listaProductos;
		this.ronda = ronda;
	}
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Date getFecha() {
		return fecha;
	}
	public void setFecha(Date fecha) {
		this.fecha = fecha;
	}
	public Date getFechaEntrega() {
		return fechaEntrega;
	}
	public void setFechaEntrega(Date fechaEntrega) {
		this.fechaEntrega = fechaEntrega;
	}
	public Usuario getUsuario() {
		return usuario;
	}
	public void setUsuario(Usuario usuario) {
		this.usuario = usuario;
	}
	public RetiroEntrega getRetiroEntrega() {
		return retiroEntrega;
	}
	public void setRetiroEntrega(RetiroEntrega retiroEntrega) {
		this.retiroEntrega = retiroEntrega;
	}
	public EstadoPedido getEstado() {
		return estado;
	}
	public void setEstado(EstadoPedido estado) {
		this.estado = estado;
	}
	public PuntoDeRetiro getDireccionRetiro() {
		return direccionRetiro;
	}
	public void setDireccionRetiro(PuntoDeRetiro direccionRetiro) {
		this.direccionRetiro = direccionRetiro;
	}
	public String getDireccionEntrega() {
		return direccionEntrega;
	}
	public void setDireccionEntrega(String direccionEntrega) {
		this.direccionEntrega = direccionEntrega;
	}
	public BigDecimal getMontoTotal() {
		return montoTotal;
	}
	public void setMontoTotal(BigDecimal montoTotal) {
		this.montoTotal = montoTotal;
	}
	public String getRangoHorario() {
		return rangoHorario;
	}
	public void setRangoHorario(String rangoHorario) {
		this.rangoHorario = rangoHorario;
	}
	public List<ProductoEncargado> getListaProductos() {
		return listaProductos;
	}
	public void setListaProductos(List<ProductoEncargado> listaProductos) {
		this.listaProductos = listaProductos;
	}
	public Ronda getRonda() {
		return ronda;
	}
	public void setRonda(Ronda ronda) {
		this.ronda = ronda;
	}
	@Override
	public String toString() {
		return "Pedido [id=" + id + ", fecha=" + fecha + ", fechaEntrega=" + fechaEntrega + ", usuario=" + usuario
				+ ", retiroEntrega=" + retiroEntrega + ", estado=" + estado + ", direccionRetiro=" + direccionRetiro
				+ ", direccionEntrega=" + direccionEntrega + ", montoTotal=" + montoTotal + ", rangoHorario="
				+ rangoHorario + ", listaProductos=" + listaProductos + ", ronda=" + ronda + "]";
	}
	

}

File: PaseoFrontEnd/src/app/modelos/pedido.ts
import { EstadoPedido } from "../utils/estado-pedido";
import { RetiroEntrega } from "../utils/retiro-entrega";
import { ProductoEncargado } from "./producto-encargado";
import { PuntoDeRetiro } from "./punto-de-retiro";
import { Ronda } from "./ronda";
import { Usuario } from "./usuario";

export class Pedido {
     id: number;
     fecha: Date;
	 fechaEntrega: Date;
	 usuario: Usuario;
	 retiroEntrega: RetiroEntrega;
	 estado: EstadoPedido;
	 direccionRetiro: PuntoDeRetiro;
	 direccionEntrega: string;
	 montoTotal: number;
	 rangoHorario: string;
	 listaProductos: ProductoEncargado[];
     ronda: Ronda|null;


	constructor( fecha: Date, fechaEntrega: Date, usuario: Usuario, retiroEntrega: RetiroEntrega, estado: EstadoPedido, direccionRetiro: PuntoDeRetiro, direccionEntrega: string, montoTotal: number, rangoHorario: string, listaProductos: ProductoEncargado[], ronda: Ronda | null ) {
        this.fecha=fecha;
        this.fechaEntrega=fechaEntrega;
        this.usuario=usuario;
        this.retiroEntrega=retiroEntrega;
        this.estado=estado;
        this.direccionRetiro=direccionRetiro;
        this.direccionEntrega=direccionEntrega;
        this.montoTotal=montoTotal;
        this.rangoHorario=rangoHorario;
        this.listaProductos=listaProductos;
        this.ronda=ronda;
    }

    public get Ronda(): Ronda|null {
        return this.ronda;
    }
    public set Ronda(value: Ronda) {
        this.ronda = value;
    }
    public get Id(): number {
        return this.id;
    }
    public set Id(value: number) {
        this.id = value;
    }
    public get Fecha(): Date {
        return this.fecha;
    }
    public set Fecha(value: Date) {
        this.fecha = value;
    }
    public get Usuario(): Usuario {
        return this.usuario;
    }
    public set Usuario(value: Usuario) {
        this.usuario = value;
    }
    
    public get FechaEntrega(): Date {
        return this.fechaEntrega;
    }
    public set FechaEntrega(value: Date) {
        this.fechaEntrega = value;
    }
    public get RetiroEntrega(): RetiroEntrega {
        return this.retiroEntrega;
    }
    public set RetiroEntrega(value: RetiroEntrega) {
        this.retiroEntrega = value;
    }
    public get Estado(): EstadoPedido {
        return this.estado;
    }
    public set Estado(value: EstadoPedido) {
        this.estado = value;
    }
    public get DireccionRetiro(): PuntoDeRetiro {
        return this.direccionRetiro;
    }
    public set DireccionRetiro(value: PuntoDeRetiro) {
        this.direccionRetiro = value;
    }
    public get DireccionEntrega(): string {
        return this.direccionEntrega;
    }
    public set DireccionEntrega(value: string) {
        this.direccionEntrega = value;
    }
    public get MontoTotal(): number {
        return this.montoTotal;
    }
    public set MontoTotal(value: number) {
        this.montoTotal = value;
    }
    public get RangoHorario(): string {
        return this.rangoHorario;
    }
    public set RangoHorario(value: string) {
        this.rangoHorario = value;
    }
    public get Productos(): ProductoEncargado[] {
        return this.listaProductos;
    }
    public set Productos(value: ProductoEncargado[]) {
        this.listaProductos = value;
    }
}
